sklearn/metrics/pairwise.py

```diff
@@ -933,21 +948,22 @@ def manhattan_distances(X, Y=None, *, sum_over_features=True):
     """
     X, Y = check_pairwise_arrays(X, Y)
 
+    if issparse(X) or issparse(Y):
+        X = csr_matrix(X, copy=False)
+        Y = csr_matrix(Y, copy=False)
+        X.sum_duplicates()  # this also sorts indices in-place
+        Y.sum_duplicates()
+
+    if sum_over_features and PairwiseDistances.is_usable_for(X, Y, metric="manhattan"):
+        return PairwiseDistances.compute(X, Y, metric="manhattan")
+
     if issparse(X) or issparse(Y):
         if not sum_over_features:
             raise TypeError(
                 "sum_over_features=%r not supported for sparse matrices"
                 % sum_over_features
             )
```

Comment:
Code coverage says that is also not reachable:

https://app.codecov.io/gh/scikit-learn/scikit-learn/pull/23958

but maybe it's because it's not up to date because of the remaining failing tests?

Not sure...

Reply:
`manhattan_distances` with `sum_over_features=False` does not return distances, looks unused, and is also unlikely used by users.

This is out of the scope of this PR, but to me the behavior for `sum_over_features=True` should be the unique one for `manhattan_distances`; in this case, `sum_over_features` would be removed this from the public API after a deprecation cycle. WDYT?

Reply:
> in this case, sum_over_features would be removed this from the public API after a deprecation cycle. WDYT?

If it is actually never used in scikit-learn, +1 for deprecation.

Reply:
I opened https://github.com/scikit-learn/scikit-learn/issues/24597 accordingly.